ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -212,14 +212,23 @@ func (scbuilder *SiteConfigBuilder) getClusterCRs(clusterId int, siteConfigTemp
 
 			// cluster-level CR
 			if kind == "ConfigMap" {
-				// For ConfigMap, add all the ExtraManifest files to it before doing further instantiation:
-				data := instantiatedCR["data"]
-				if data != nil {
-					for k, v := range data.(map[string]interface{}) {
-						extraManifestMap[k] = v
+				configMapMetadata := mapSourceCR["metadata"].(map[string]interface{})
+				configMapName := configMapMetadata["name"].(string)
+				if strings.Contains(configMapName, "SiteConfigMap") {
+					// If there is no siteConfigMap specified, don't create the corresponding CR.
+					if cluster.SiteConfigMapIsEmpty() {
```

Comment:
If the new siteConfigMap is empty in the siteConfig CR, I don't want it to be added to the clusterCRs. If it's not empty, then it should directly be added to the clusterCRs without any other processing.

Reply:
SiteConfigMap could be a pointer to SiteConfigMap struct, in that case any empty SiteConfigMap would return nil value and probably it can make the logic easier. Just a thought.

Reply:
The CM logic is fine, I mean the if checks something like below more clean code
`
if !strings.Contains(configMapName, "SiteConfigMap") {

// For ConfigMap, add all the ExtraManifest files to it before doing further instantiation:
  data := instantiatedCR["data"]
  if data != nil {
     for k, v := range data.(map[string]interface{}) {
		extraManifestMap[k] = v
    }
  }
 instantiatedCR["data"] = extraManifestMap
}

`